PairLinCorr/pairlincorr.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from sklearn.model_selection import train_test_split
+
+MODEL_DIR = 'model_dir_py/'
+
+# import data
+data = pd.read_csv(
+    'input.csv', dtype={
+        'item1': str,
+        'value1': float,
+        'item2': str,
+        'value2': float
+    })
+print('data size: {}'.format(len(data)))
+
+x = data[['item1', 'value1', 'item2', 'value2']]
+y = data['value2'] - data['value1']
+
+# split training and test sets
+x_train, x_test, y_train, y_test = train_test_split(
+    x, y, test_size=0.2, random_state=1)
+
+# count number of unique item values
+list_items = x_train['item1'].append(
+    x_train['item2']).drop_duplicates().tolist()
+n_items = len(list_items)
+print("number of unique items: {}".format(n_items))
+
+# columns
+item1_col = tf.feature_column.indicator_column(
+    tf.feature_column.categorical_column_with_vocabulary_list(
+        'item1', vocabulary_list=list_items))
+value1_col = tf.feature_column.numeric_column('value1')
+item2_col = tf.feature_column.indicator_column(
+    tf.feature_column.categorical_column_with_vocabulary_list(
+        'item2', vocabulary_list=list_items))
+value2_col = tf.feature_column.numeric_column('value2')
+
+
+# build input function for training
+input_fn_train = tf.estimator.inputs.pandas_input_fn(
+    x=x_train,
+    y=y_train,
+    batch_size=100,
+    num_epochs=3,
```

Comment:
Higher `num_epochs` will reduce loss but we keep it to 3 here for convenience of development.